### PR TITLE
improved readme & fixed issue #5

### PR DESCRIPTION
**Pull request by kacaleksandra**

### CODE REVIEW

File: backend/src/notes/notes.service.ts [hunk 1]
```diff
@@ -66,25 +90,59 @@ export class NotesService {
...
+              categoryId,
+            },
+          });
+        }
+      } else {
+        // Jeśli categoryId nie jest przesłane, usuwamy wszelkie powiązania w tabeli pośredniczącej
+        await this.prisma.noteCategories.deleteMany({
+          where: { noteId: id },
+        });
+      }
```
Comment (bartekpacia) [inline backend/src/notes/notes.service.ts:126]: Polish comments?

:-(

File: backend/src/notes/notes.service.ts [hunk 2]
```diff
@@ -66,25 +90,59 @@ export class NotesService {
...
   }
 
-  async remove(userId: number, id: number) {
+  async remove(
+    userId: number,
+    id: number,
+  ): Promise<{ success: boolean; message?: string }> {
     const note = await this.prisma.notes.findUnique({
       where: { id, userId },
+      include: { NoteCategories: true, Reminders: true }, // Pobieramy powiązane kategorie
```
Comment (bartekpacia) [inline backend/src/notes/notes.service.ts:145]: polish

comment

File: backend/src/notes/notes.service.ts [hunk 3]
```diff
@@ -96,23 +154,39 @@ export class NotesService {
     }
 
     try {
+      // Usuwamy powiązania z tabeli pośredniczącej, jeśli istnieją
```
Comment (bartekpacia) [inline backend/src/notes/notes.service.ts:157]: aaaaaaa

**Review by bartekpacia — CHANGES_REQUESTED** [review]
Please fix comments in Polish

**Review by bartekpacia — APPROVED** [review]
lessgo

File: backend/src/categories/categories.service.ts [hunk 4]
```diff
@@ -42,24 +42,27 @@ export class CategoriesService {
...
-    } catch (error) {
-      throw new InternalServerErrorException('Failed to delete category');
-    }
+      try {
+        await prisma.noteCategories.deleteMany({ where: { categoryId } });
+        await prisma.categories.delete({ where: { id: categoryId } });
+      } catch (error) {
+        throw new InternalServerErrorException('Failed to delete category');
+      }
+    });
```
[the remark this answers is not in the document]
Comment (kacaleksandra) [reply backend/src/categories/categories.service.ts:65]: no error, the connections to categories in notes will be deleted

File: backend/src/notes/notes.service.ts (hunk 1 above)
Comment (kacaleksandra) [reply backend/src/notes/notes.service.ts:126]: yes :-)

File: backend/src/notes/notes.service.ts (hunk 3 above)
Comment (kacaleksandra) [reply backend/src/notes/notes.service.ts:157]: bbbbbbb

File: backend/src/notes/notes.service.ts (hunk 2 above)
Comment (kacaleksandra) [reply backend/src/notes/notes.service.ts:145]: polish

comment

**Pull request merged**